flow-typed/ka-types.js

```diff
@@ -0,0 +1,8 @@
+// @flow
+
+// Marks all properties of an object optional.
+//
+// We use this instead of $Shape<T> because spreading $Shape<$Exact<T>> ends up
+// spreading $Exact<T> instead, losing the optionality of keys.
+// https://github.com/facebook/flow/issues/6906#issuecomment-453439922
+declare type Partial<T> = $ReadOnly<$Rest<T, {...}>>;
```

Comment:
I'm not sure if it makes sense to introduce a type like this in a library.

Reply:
Do we still need it or can we use Flow's own `$Partial`?

Reply:
I didn't realize that Flow had added that.  I'll switch to use that instead.